### PR TITLE
Fix bug caused by builtin generics not converting str to ForwardRef

### PR DESCRIPTION
**Pull request by dmontagu**
Selected Reviewer: @hramezani

### CODE REVIEW

File: tests/test_edge_cases.py [hunk 1]
```diff
@@ -2346,3 +2346,23 @@ class Square(AbstractSquare):
         ),
     ):
         Square(side=1.0)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 9), reason='cannot use list.__class_getitem__ before 3.9')
+def test_generic_wrapped_forwardref():
+    class Operation(BaseModel):
```
Comment (dmontagu) [inline tests/test_edge_cases.py:2353]: Without the changes in this PR, you actually get `{'type': 'PathItem'}` as an intermediate output of `GenerateSchema._generate_schema` (i.e., as an attempt at building a CoreSchema), which unsurprisingly makes class creation fail.

**Comment by dmontagu** [comment]
please review

File: pydantic/_internal/_generate_schema.py [hunk 2]
```diff
@@ -279,12 +279,14 @@ def _generate_schema(self, obj: Any, from_dunder_get_core_schema: bool) -> core_
         """
         Recursively generate a pydantic-core schema for any supported python type.
         """
-        if isinstance(obj, str):
-            return {'type': obj}  # type: ignore[return-value,misc]
```
Comment (dmontagu) [inline pydantic/_internal/_generate_schema.py:283]: For context, without the changes in this PR, this line was not covered in our current test suite (I replaced it with assert False and all tests were passing).

**Comment by samuelcolvin** [comment]
thank you.

**Pull request merged**